crates/e2e/tests/e2e/liquidity.rs

```diff
@@ -206,47 +196,46 @@ async fn zero_ex_liquidity(web3: Web3) {
         .unwrap();
 
     let sell_token_balance_after = token_usdc
-        .balance_of(trader_a.address())
+        .balance_of(trader.address())
         .call()
         .await
         .unwrap();
     let buy_token_balance_after = token_usdt
-        .balance_of(trader_a.address())
+        .balance_of(trader.address())
         .call()
         .await
         .unwrap();
 
     assert!(sell_token_balance_before > sell_token_balance_after);
-    assert!(buy_token_balance_after >= buy_token_balance_before + to_wei_with_exp(500, 6));
+    assert!(buy_token_balance_after >= buy_token_balance_before + to_wei_with_exp(amount, 6));
 }
 
 fn orders_query_handler(
     query: &OrdersQuery,
     order_creation: OrderCreation,
     zeroex_maker: TestAccount,
-    zeroex_taker: TestAccount,
     zeroex_addr: H160,
     gpv2_addr: H160,
     chain_id: u64,
 ) -> Result<Vec<OrderRecord>, ZeroExResponseError> {
     if query.sender == Some(gpv2_addr) {
         let typed_order = Eip712TypedZeroExOrder {
-            maker_token: order_creation.sell_token,
-            taker_token: order_creation.buy_token,
-            maker_amount: order_creation.sell_amount.as_u128() * 2,
-            taker_amount: order_creation.buy_amount.as_u128() * 2,
+            maker_token: order_creation.buy_token,
+            taker_token: order_creation.sell_token,
+            maker_amount: order_creation.buy_amount.as_u128() * 3,
+            taker_amount: order_creation.sell_amount.as_u128() * 2,
```

Comment:
This multiplication is required to pass [this check](https://github.com/cowprotocol/services/blob/baseline-solver-limit-orders-support/crates/solvers/src/domain/solution.rs#L203-L205). Not sure it is 100% valid.

Reply:
🤔 to me it seems that the inequality in the check should be a `<=` instead of `<`.

Why even multiply with 2 (maybe add a comment)?

Reply:
Maybe instead of adjusting the amount with these factors it's a better idea to make the order partially fillable. That way we don't have to have a comment for all the factors but only when we assert that the traded amounts look good in the end.